packages/sfui/shared/scrollable/scrollable.ts

```diff
@@ -1,111 +1,146 @@
-import type { SfScrollableOptions, SfScrollableCallbackData } from './types';
+import { type ScrollableOptions, type SfScrollableOnScrollData, SfScrollableDirection } from './types';
+import { isReduceMotionEnabled } from '../utils';
 
-const defaultSfScrollableOptions: SfScrollableOptions = {
-  reduceMotion: false,
-  snap: false,
-  drag: undefined,
-  vertical: false,
+const defaultScrollableOptions: ScrollableOptions = {
+  reduceMotion: isReduceMotionEnabled,
+  direction: SfScrollableDirection.horizontal,
 };
 
-const DRAGGING_CLASS = 'sf-scrollable__container--dragging';
-
-export default class SFScrollable {
+export default class Scrollable {
   public container: HTMLElement;
-  private options: SfScrollableOptions;
+  private options: ScrollableOptions;
   private debounceId?: ReturnType<typeof setTimeout>;
-  private isDragged?: boolean;
   private dragScrollX: number;
   private dragScrollLeft: number;
   private dragScrollY: number;
   private dragScrollTop: number;
 
-  private scrollListenerInstance: (event: Event) => void = () => undefined;
-  private resizeListenerInstance: () => void = () => undefined;
-  private mouseDownListenerInstance: (event: MouseEvent) => void = () => undefined;
-  private mouseUpListenerInstance: (event: MouseEvent) => void = () => undefined;
-  private mouseMoveListenerInstance: (event: MouseEvent) => void = () => undefined;
-  private mouseLeaveListenerInstance: (event: MouseEvent) => void = () => undefined;
+  private isDraggedData: boolean = false;
+  public get isDragged() {
+    return this.isDraggedData;
+  }
+  public set isDragged(newValue: boolean) {
+    if (newValue === this.isDraggedData) return;
+    this.isDraggedData = newValue;
+    this.options.onDragChange?.({ isDragged: this.isDraggedData });
+  }
```

Comment:
I have to admit that `drag` option does not work well with css scroll features and I would consider to drop this feature there as is error prone.
There are many (but heavier) alternatives on the market, so this solution should not compete with them, but should be lightweight, reliable & SSR friendly alternative (and ofc easy to maintain).

Reply:
You are most likely right, there are better library but since we have this and its light, we can have drag in basic form, maybe we will extend it in future

Reply:
BTW, AFAIK `mouse` events don't work on mobile and elements will fallback to native touch events (`drag` events will be always available for mobile), so, if you need to cover some special cases on mobile (or provide consistency for all devices), maybe should use [`pointer events`](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events) instead and disable native events.

As I mentioned above, `drag` feature is tricky.